### ISSUE TITLE
"images to the dataset as negative examples" !

### ISSUE THREAD

**Issue by salihkaragoz** [yhenon/pytorch-retinanet#139]
Hi, Great Work!
I reviewed and debugged the code in detail based on the Readme Note. ("Some images may not contain any labeled objects. To add these images to the dataset as negative examples, add an annotation where x1, y1, x2, y2 and class_name are all empty:")

When you put a line without bbox values and class name to the CSV file, the dataloader first prepares ground truth as [0,0,0,0,0]. 
(https://github.com/yhenon/pytorch-retinanet/blob/master/retinanet/dataloader.py#L224-L229)

Then in this line, the dataloader_train makes the ground truth to [-1,-1,-1,-1,-1]. (https://github.com/yhenon/pytorch-retinanet/blob/master/retinanet/dataloader.py#L322-L334)

In the loss function, if the ground truth is -1 it returns loss value 0 for both classification and regression. (https://github.com/yhenon/pytorch-retinanet/blob/master/retinanet/losses.py#L47)

I also observed in the train_dataloader(https://github.com/yhenon/pytorch-retinanet/blob/master/train.py#L121)
the loss value of the non-object image returns zero. 

so as a result, Correct me if I am wrong,  
I think we can say that the images without any object inside don't help to the training while using the CSV generator. 
Thanks

**UPDATE**
If the batch size 1, the above is valid. Otherwise no.
